source/dubregistry/web.d

```diff
@@ -595,6 +623,30 @@ class DubRegistryFullWebFrontend : DubRegistryWebFrontend {
 		redirect("/my_packages/"~_packname);
 	}
 
+	@auth @path("/my_packages/:packname/set_logo")
+	void postSetLogo(scope HTTPServerRequest request, string _packname, User _user)
+	{
+		enforceUserPackage(_user, _packname);
+		const(FilePart) logo = request.files.get("logo");
+		enforceBadRequest(logo != FilePart.init);
+		auto renamed = NativePath.fromString(logo.tempPath.toString ~ logo.filename.name.extension);
```

Comment:
`NativePath.fromString` should ideally accept a range...

Reply:
Definitely wouldn't hurt from a convenience POV, but it stores a `string` inside anyway, so that it couldn't eliminate the allocation.

Reply:
or maybe some rename filename/last segment? (or any segment even)

Reply:
> so that it couldn't eliminate the allocation.

Hmm, well I guess it could be templatized (would be nice with the hopefully soon upcoming RCString), but it's a different story entirely.
Anyhow I opened an issue, s.t. it's not totally forgotten (https://github.com/vibe-d/vibe.d/issues/2119).